lms/services/oauth_http.py

```diff
@@ -14,9 +15,12 @@ class _OAuthAccessTokenErrorResponseSchema(RequestsResponseSchema):
 class OAuthHTTPService:
     """Send OAuth 2.0 requests and return the responses."""
 
-    def __init__(self, http_service, oauth2_token_service):
+    def __init__(
+        self, http_service, oauth2_token_service, service: Service = Service.LMS
+    ):
         self._http_service = http_service
         self._oauth2_token_service = oauth2_token_service
+        self.service = service
```

Comment:
We do have some other instances where one service creates a local instance of another service (eg. `HTTPService`) rather than using the "shared" request-wide one, but those existing cases just call the service class directly. I could have done that here, but it seems useful to reuse the factory's dependency lookup.

The rationale for having `service` be an instance property is to enforce that it is consistent across different requests and avoid a situation where some code path forgets to set it.